lib/shopify_cli/theme/extension/dev_server/hooks/file_change_hook.rb

```diff
@@ -6,34 +6,59 @@ module Extension
       class DevServer
         module Hooks
           class FileChangeHook
-            def initialize(ctx, extension:)
+            attr_reader :ctx, :extension, :syncer, :streams
+
+            def initialize(ctx, extension:, syncer:)
               @ctx = ctx
               @extension = extension
+              @syncer = syncer
             end
 
             def call(modified, added, removed, streams: nil)
               @streams = streams
-              files = (modified + added)
-                .map { |f| @extension[f] }
-                .reject(&:liquid_css?)
-              deleted_files = removed
-                .map { |f| @extension[f] }
 
-              reload_page(removed) unless deleted_files.empty?
-              hot_reload(files) unless files.empty?
+              modified = paths(modified).select { |file| @extension.extension_file?(file) }
+              added = paths(added).select { |file| @extension.extension_file?(file) }
+              removed = paths(removed)
```

Comment:
in theory could we check if the removed file are extension files by comparing the extension root with the removed file root?

Reply:
I believe that check would always return true as the [`FileSystemListener`](https://github.com/Shopify/shopify-cli/blob/fix-2586/lib/shopify_cli/theme/extension/dev_server/watcher.rb#L21) triggers events based on the extension root. I'm not sure if I'm missing something tho, please let me know :)

(I will proceed with the merge, but I can apply that enhancement in a new PR if I missed something)

Thanks for the comment, @mgmanzella! :)